src/neurotechdevkit/scenarios/_scenario_2.py

```diff
@@ -51,20 +26,22 @@ class _Scenario2Mixin:
         https://asa.scitation.org/doi/pdf/10.1121/10.0013426
     """
 
-    @property
-    def _material_layers(self: _Scenario2MixinProtocol) -> list[tuple[str, Struct]]:
-        return [
-            ("water", materials.water),
-            ("skull", materials.cortical_bone),
-            ("brain", materials.brain),
-        ]
+    material_layers = [
+        "water",
+        "cortical_bone",
+        "brain",
+    ]
 
     @property
     def _material_outline_upsample_factor(self) -> int:
         return 4
 
+    def _get_material_masks(self) -> Mapping[str, npt.NDArray[np.bool_]]:
+        """Will be implemented by the subclass."""
+        raise NotImplementedError()
+
     def _compile_scenario_2_problem(
-        self: _Scenario2MixinProtocol, extent: npt.NDArray[np.float_]
+        self, extent: npt.NDArray[np.float_]
```

Comment:
`c_freq` below is staying constant instead of updating with the user defined value.